src/sink.ts

```diff
@@ -153,23 +151,16 @@ class Sink {
   delete(gaxOptions?: CallOptions): Promise<DeleteResponse>;
   delete(callback: DeleteCallback): void;
   delete(gaxOptions: CallOptions, callback: DeleteCallback): void;
-  delete(gaxOptions?: CallOptions|DeleteCallback, callback?: DeleteCallback):
-      Promise<DeleteResponse>|void {
-    if (typeof gaxOptions === 'function') {
-      callback = gaxOptions as DeleteCallback;
-      gaxOptions = {};
-    }
+  async delete(gaxOptions?: CallOptions|
+               DeleteCallback): Promise<DeleteResponse> {
+    this.logging.projectId = await this.logging.auth.getProjectId();
```

Comment:
Instead of updating the instance variable here, let's just use a local variable to hold the project ID. I think the pattern should just be to get the project ID from the auth client when needed, instead of relying on `this.logging.projectId`.

Reply:
Done